packages/alfa-style/src/property/mask.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  Token,
+  List,
+  type Parser as CSSParser,
+  Position,
+  Box,
+  Keyword,
+} from "@siteimprove/alfa-css";
+import { Parser } from "@siteimprove/alfa-parser";
+import type { Slice } from "@siteimprove/alfa-slice";
+import { Option } from "@siteimprove/alfa-option";
+
+import { Shorthand } from "../shorthand.js";
+
+import { MaskReference } from "./mask-image.js";
+import { BgSize } from "./mask-size.js";
+import { RepeatStyle } from "./mask-repeat.js";
+import { CompositingOperator } from "./mask-composite.js";
+import { MaskingMode } from "./mask-mode.js";
+import { MaskPosition } from "./mask-position.js";
+import { MaskClip } from "./mask-clip.js";
+import { MaskOrigin } from "./mask-origin.js";
+
+const {
+  doubleBar,
+  either,
+  map,
+  option,
+  pair,
+  right,
+  delimited,
+  separatedList,
+} = Parser;
+
+const slash = delimited(option(Token.parseWhitespace), Token.parseDelim("/"));
+
+const parsePosAndSize: CSSParser<[Position, Option<BgSize>]> = pair(
+  Position.parse(/* legacySyntax */ true),
+  option(right(slash, BgSize.parse)),
+);
+
+/**
+ * {@link https://drafts.fxtf.org/css-masking/#typedef-mask-layer}
+ *
+ * @privateRemarks
+ * As of December 2024 the specification uses the <geometry-box> type in the shorthand defintion
+ * whereas the longhands `mask-clip` and `mask-origin` uses <coord-box>.
+ * These are not the same types - <coord-box> does not have `margin-box`.
+ * Chrome and Firefox does not, at time of writing, allow `margin-box` in the shorthand.
+ * Therefore we assume that the discrepancy is a spec-bug and that the intended type is <coord-box>.
+ */
+const maskLayer: CSSParser<
+  [
+    MaskReference | undefined,
+    Position | undefined,
+    BgSize | undefined,
+    RepeatStyle | undefined,
+    Box.CoordBox | undefined,
+    Box.CoordBox | Keyword<"no-clip"> | undefined,
+    CompositingOperator | undefined,
+    MaskingMode | undefined,
+  ]
+> = map(
+  doubleBar<
+    Slice<Token>,
+    [
+      MaskReference,
+      [Position, Option<BgSize>],
+      RepeatStyle,
+      Box.CoordBox,
+      Box.CoordBox | Keyword<"no-clip">,
+      CompositingOperator,
+      MaskingMode,
+    ],
+    string
+  >(
+    Token.parseWhitespace,
+    MaskReference.parse,
+    parsePosAndSize,
+    RepeatStyle.parse,
+    Box.parseCoordBox,
+    either(Box.parseCoordBox, Keyword.parse("no-clip")),
+    CompositingOperator.parse,
+    MaskingMode.parse,
+  ),
+  ([image, posAndSize, repeat, box1, box2, composite, mode]) => {
+    const [pos, size] =
+      posAndSize !== undefined
+        ? [posAndSize[0], posAndSize[1].getOr(undefined)]
+        : [undefined, undefined];
+
+    const origin = box1;
+    const clip = box2 ?? box1;
+
+    return [image, pos, size, repeat, origin, clip, composite, mode] as const;
+  },
+);
+
+const parseList = separatedList(
+  maskLayer,
+  delimited(option(Token.parseWhitespace), Token.parseComma),
+);
```

Comment:
**Question:** Isn't that a `List.parseCommaSeparated`?

Reply:
`List.parseCommaSeparated` only accepts a parser whose result is an `alfa-css` `Value`, but the result of the shorthand parser is not a `Value`. The same applies in [background.ts](https://github.com/Siteimprove/alfa/blob/092e524423ea29e494077c001e88981f05aa19a8/packages/alfa-style/src/property/background.ts#L199-L202). Maybe the `parse[Comma|Space]Separated` functions can be lifted to `alfa-parser` for better reuse, but maybe something for later.

Reply:
Ah, good point.
We can't really lift these to `alfa-parser` more than the existing `separatedList`, because the `Token` are an `alfa-css` thing. Best case, we may refactor `delimited(option(Token.parseWhitespace), Token.parseComma)` into a single parser since it's used a lot, but that's a side-improvement for another time.